stdlib/public/core/UnicodeScalarProperties.swift

```diff
@@ -692,15 +692,16 @@ extension Unicode.Scalar.Properties {
   /// all current case mappings. In the event more space is needed, it will be
   /// allocated on the heap.
   internal func _applyMapping(_ u_strTo: _U_StrToX) -> String {
-    // TODO(String performance): Stack buffer first and then detect real count
     let count = 64
-    var array = Array<UInt16>(repeating: 0, count: count)
-    let len: Int = array.withUnsafeMutableBufferPointer { bufPtr in
-      return _scalar.withUTF16CodeUnits { utf16 in
+    var array = _FixedArray16<UInt64>(allZeros: ()) // 16 * UInt64 take 128 bytes, same as 64 * UInt16
+    let string: String = array.withUnsafeMutableBufferPointer { bufPtrUInt64 in
+      let bufPtrUInt16: UnsafeMutablePointer<UInt16> = UnsafeMutableRawPointer(bufPtrUInt64.baseAddress._unsafelyUnwrappedUnchecked)
```

Comment:
Nit: 80-column lines, here and below :)
Nit: `bufPtrUInt16` isn't a buffer pointer; consider renaming.

More substantially, I wonder if it would be more preferable to avoid the outer typed pointer. `_FixedArray16` first assumes that the memory is bound to `UInt64`, and then it is rebound to `UInt16`. Instead, it could be, I think:

```swift
let string: String = withUnsafeMutableBytes(of: &array.storage) { rawBuffer in
  let ptr = rawBuffer.baseAddress._unsafelyUnwrappedUnchecked
    .bindMemory(to: UInt16.self, capacity: count)
  // ...
```